solidpy/Environment.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+import math
+import scipy.constants as const
+
+
+class Environment:
+    def __init__(
+        self,
+        atmospheric_pressure=const.atmosphere,
+        air_density=1.225,
+        latitude=0,
+        gravity=None,
+        ellipsoidal_model=False,
+    ):
+        self.atmospheric_pressure = atmospheric_pressure
+        self.air_density = air_density
+        self.latitude = latitude
+        self.gravity = self.evaluate_gravity(gravity, latitude, ellipsoidal_model)
+
+    def evaluate_gravity(self, gravity, latitude, ellipsoidal_model):
+        if gravity is None and ellipsoidal_model:
+            # Somigliana approximation for ellipsoidal gravity
+            gravity = 9.78032533590389 * (
+                (1 + 0.001931852652458 * (math.sin(latitude)) ** 2)
+                / (math.sqrt(1 - 0.006694379990141 * (math.sin(latitude)) ** 2))
```

Comment:
Very interesting feature! Results will probably vary very little, but it's good to start early when trying out these kind of models.

Do you plan on adding further approximations regarding height (altitude) effects on theoretical gravity?

Reply:
The main purpose of this feature is to get more precise data when it is known to be available by a simple model such as Somigliana's.

In fact, as it was completely expected, the difference between utilizing standard gravity and this model are low: out rail velocity variance was about 0.1 m/s in worst case scenario. Of course our simulation has many other factors that introduce higher magnitude variances: the eq. diff solver model, burn rate tables, air conditions that can not be simulated (wind, humidity) etc. 

Hence, this model of theoretical gravity stands for the ideal that, if we can get theoretically more accurate results with reasonable effort, there is no reason so as not to do it; even though it will almost undoubtedly have no practical meaning.

About the addition of height dependency, I believe it is the same scenario, but with even lower magnitude. Personally, I would add it as an optional parameter for the user, since it is also a simple formula (in the sense that it does not slow down code), as it can be seen in the link below.

https://en.wikipedia.org/wiki/Theoretical_gravity?wprov=sfla1 
(Cassinis or WELMEC models)

Furthermore, I have some experience with air pressure and density models and I believe that these dependencies are more relevant than the theoretical gravity models, due to the fact that they both influence our results: the pressure is relevant for nozzle expansion and thrust calculation, while the density is present in drag evaluation. The results will also most likely be of theoretical value, with little practical meaning, but it may be relevant, specially if the rail is long.

Reply:
Finally, I completed the gravity model with height dependency. It was also a simple formula.

Combining all these changes, the maximum delta of out rail velocity is about 3% in a worst case scenario that we launched at La Paz with long rail. I think we can call this resolved for now.